pkg/controllers/options.go

```diff
@@ -41,3 +42,11 @@ func syncOnlyChangedObjects(option controllerContextType) bool {
 	}
 	return false
 }
+
+// WebhookImpersonation returns a ImpersonationConfig that can be used for impersonating the webhook's sudo account and bypass validation.
+func WebhookImpersonation() rest.ImpersonationConfig {
+	return rest.ImpersonationConfig{
+		UserName: "system:serviceaccount:cattle-system:rancher-webhook-sudo",
```

Comment:
Do we even need a special user for this now?

Reply:
Are you asking if we need the user because we are using system:masters. Or
Are you asking if we need this change for Rancher to work today?

For the first question.
My answer is yes, we need a special user. Since this indicates to the Webhook not to perform any validation or mutations and return an allowed response. If we had the webhook only look for the system:masters group, then we have admins bypassing desired mutation when they do not want to.

For the second question.
The answer is no, We do not need these changes until the RT and GR validation is migrated to the webhook.